packages/linear-pools/contracts/silo-linear-pool/SiloExchangeRateModel.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-pool-utils/contracts/lib/ExternalCallLib.sol";
+
+import "./interfaces/ISilo.sol";
+import "./interfaces/IInterestRateModel.sol";
+import "./SiloHelpers.sol";
+
+// Created in order to decrease exchange rate timelag when wrapping and unwrapping tokens
+// between Silo Linear Pools and the Silo Protocol
+contract SiloExchangeRateModel {
+    /**
+     * @dev This function is similar to _accrueInterest function in the Silo's BaseSilo.sol contract
+     * which is used to update state data that is necessary
+     */
+    function calculateExchangeValue(
+        uint256 amount,
+        IShareToken shareToken,
+        ISilo.AssetStorage memory assetStorage,
+        ISilo.AssetInterestData memory interestData
+    ) external view returns (uint256) {
+        // rcomp: compound interest rate from the last update until now
+        // Use getCompoundInterestRate() instead of getCompoundInterestRateAndUpdate becasue we are operating within
+        //      a view function and cannot manipute state
+        try
+            getModel(shareToken.silo(), shareToken.asset()).getCompoundInterestRate(
+                address(shareToken.silo()),
+                shareToken.asset(),
+                block.timestamp
+            )
+        returns (uint256 rcomp) {
+            uint256 accruedInterest = (assetStorage.totalBorrowAmount * rcomp) / 1e18;
```

Comment:
Check GearboxLinearPoolRebalancer.sol. If you import FixedPoint from `@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol` and use this line in the beginning of your class `using FixedPoint for uint256;`, then these divisions by 1e18 are not needed, because you use mulDown function. Example:

`uint256 accruedInterest = assetStorage.totalBorrowAmount.mulDown(rcomp);`

MulDown already removes 18 decimals at the end of the operation, in a standard way.

Reply:
Thank you for mentioning this package, I will incorporate it within the exchange rate model.